DirectGuiExtension/DirectDiagram.py

```diff
@@ -15,10 +15,10 @@ class DirectDiagram(DirectFrame):
     def __init__(self, parent = None, **kw):
         optiondefs = (
             # Define type of DirectGuiWidget
-            ('data',           [],          self.refresh),
-            ('numPosSteps',     0,          self.refresh),
+            ('data',           [],  self.refresh),
+            ('numPosSteps',     1,          self.refresh),
```

Comment:
It'd be better to keep this at 0 since that value is used to automatically calculate the steps from the given data and would break the default behavior. There should probably be a check added to the refresh method to not crash if no data is given.
Same goes for the numNegSteps.

Reply:
Sure, I can fix that and add a check in refresh.